### PR TITLE
docs(tRPC): Switch to new `setData` signature

### PR DESCRIPTION
**Pull request by bennettdams**
## ✅ Checklist

- [x] I have followed every step in the [contributing guide](https://github.com/t3-oss/create-t3-app/blob/main/CONTRIBUTING.md) (updated 2022-10-06).
- [x] The PR title follows the convention we established [conventional-commit](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] I performed a functional test on my final commit

---

## Changelog

The signature for `setData` has changed. This commit will do two things:
- Provide `undefined` for the input argument (mandatory).
- Use the updater function (`old => [...old, newPost ]`) instead of the `prevData` directly. This makes it also align with the React Query docs:
  
  image

### CODE REVIEW

File: www/src/pages/en/usage/trpc.md [hunk 1]
```diff
@@ -258,10 +258,13 @@ const MyComponent = () => {
     async onMutate(newPost) {
       // Cancel outgoing fetches (so they don't overwrite our optimistic update)
       await utils.post.list.cancel();
+
       // Get the data from the queryCache
       const prevData = utils.post.list.getData();
```
Comment (c-ehrlich) [inline www/src/pages/en/usage/trpc.md:263]: Not sure about the API changes, Julius can probably give a comment on that. But if we're not using `prevData` anymore then we don't need it here.

Comment (bennettdams) [reply www/src/pages/en/usage/trpc.md:263]: I let `prevData` like it is, as one can use it to put it into the context by returning it from `onMutate`. That way, you can use the previous data to roll back (see `onError` below) on a failed mutation.

Comment (bennettdams) [reply www/src/pages/en/usage/trpc.md:263]: Oh, I just realized there's also another old `setData` call in `onError`. Will add another commit.

File: www/src/pages/en/usage/trpc.md [hunk 2]
```diff
@@ -258,16 +258,19 @@ const MyComponent = () => {
...
+
       // Optimistically update the data with our new post
-      utils.post.list.setData([...prevData, newPost ]);
+      utils.post.list.setData(undefined, old => [...old, newPost ]);
```
Comment (juliusmarminge) [inline www/src/pages/en/usage/trpc.md:266]:
```suggestion
      utils.post.list.setData(undefined, (old) => [...old, newPost]);
```

**Pull request merged**